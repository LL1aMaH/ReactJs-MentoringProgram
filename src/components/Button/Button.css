.button {
  display: flex;
  align-items: center;
  padding: 0.125rem 0.35rem;
  font-size: 0.875rem;
  line-height: 1rem;
  color: var(--primary-font-color);
  background-color: transparent;
  border-radius: 2px;
  border: none;
  cursor: pointer;
  height: auto;
  outline: none;
}

.disabled {
  background-color: var(--disabled-color);
  border-color: var(--disabled-color);
  color: var(--disabled-color);
  cursor: default;
}
