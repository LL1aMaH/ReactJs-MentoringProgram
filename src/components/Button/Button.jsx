import React, { memo } from 'react';
import classnames from 'classnames';

import styles from '../../../styles/Button.module.scss'


export const Button = memo(function Button({
  className,
  children,
  onClick,
  tabIndex = 0,
  disabled = false,
  testId,
  title,
  type,
  ...restProps
}) {
  return (
    <button
      {...restProps}
      className={classnames(className, styles.button, {
        [styles.disabled]: disabled,
      })}
      type={type}
      onClick={onClick}
      tabIndex={tabIndex}
      disabled={disabled}
      data-testid={testId}
      title={title}
    >
      {children}
    </button>
  );
});
