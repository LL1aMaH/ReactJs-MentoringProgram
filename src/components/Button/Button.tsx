/* eslint-disable react/button-has-type */
/* eslint-disable react/jsx-props-no-spreading */
import React, { ReactNode, ButtonHTMLAttributes, memo } from 'react';
import classnames from 'classnames';

import styles from './Button.css';

export type TButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  children: ReactNode;
  testId?: string;
  type?: 'submit' | 'reset' | 'button';
};

export const Button = memo(function Button({
  className,
  children,
  onClick,
  tabIndex = 0,
  disabled = false,
  testId,
  title,
  type,
  ...restProps
}: TButtonProps): JSX.Element {
  return (
    <button
      {...restProps}
      className={classnames(className, styles.button, {
        [styles.disabled]: disabled,
      })}
      type={type}
      onClick={onClick}
      tabIndex={tabIndex}
      disabled={disabled}
      data-testid={testId}
      title={title}
    >
      {children}
    </button>
  );
});
