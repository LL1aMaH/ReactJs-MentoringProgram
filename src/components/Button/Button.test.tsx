import React from 'react';
import { render } from '@testing-library/react';

import { Button } from './Button';

describe('Button', () => {
  it('renders correctly', () => {
    const getComponent = () => render(<Button onClick={jest.fn}>Button text</Button>);

    expect(getComponent().asFragment()).toMatchSnapshot();
  });
});
