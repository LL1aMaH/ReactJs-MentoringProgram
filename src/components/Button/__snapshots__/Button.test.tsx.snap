// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="button"
    tabindex="0"
  >
    Button text
  </button>
</DocumentFragment>
`;
